### ISSUE TITLE
Same schema cannot be referred twice in message definition otherwise the generation fails

### ISSUE THREAD

**Issue by CoolTomatos** [ZenWave360/zenwave-sdk#1]
I am trying to use the `zenwave-sdk-maven-plugin` with generator `jsonschema2pojo` and I have noticed a strange behaviour that I think it is a bug:

```yaml
asyncapi: '2.6.0'
info:
  title: Account Service
  version: 1.0.0
  description: This service is in charge of processing user signups
channels:
  user/signedup:
    subscribe:
      operationId: user-signedup
      message:
        $ref: '#/components/messages/UserSignedUp'
components:
  messages:
    UserSignedUp:
      payload:
        type: object
        properties:
          displayName:
            type: string
            description: Name of the user
          email:
            $ref: "#/components/schemas/Email"
            description: Email of the user
          backupEmail:
            description: something else
            $ref: "#/components/schemas/Email"
  schemas:
    Email:
      type: string
      enum:
        - Google
        - Yahoo
```    
Given the above definition, generation fails. And if I comment out the `backupEmail` property, the generation succeed.

**Comment by ivangsa** [comment]
Hi @CoolTomatos 

thanks for reporting this, and my apologies becasuse I didn't see it before...

Indeed the upstream library https://www.jsonschema2pojo.org/ is not able to process this schema...

The resolved schema is

```json
{
    "type": "object",
    "properties": {
        "displayName": {
            "type": "string",
            "description": "Name of the user"
        },
        "email": {
            "type": "string",
            "enum": [
                "Google",
                "Yahoo"
            ],
            "javaType": "io.example.integration.test.with_schemas.model.Email",
            "x--schema-name": "Email"
        },
        "backupEmail": {
            "type": "string",
            "enum": [
                "Google",
                "Yahoo"
            ],
            "javaType": "io.example.integration.test.with_schemas.model.Email",
            "x--schema-name": "Email"
        }
    }
}
```

you can test it on their preview site...

I will work now with upstream library to solve this issue...

Thanks for taking the time to report this, and my apologies again for not seen it earlier

**Comment by ivangsa** [comment]
A fix for this was released as part of version 1.0.5

**Comment by CoolTomatos** [comment]
Hi @ivangsa, thanks for the quick fix!
